pypdf/xmp.py

```diff
@@ -231,21 +230,8 @@ def rdfRoot(self) -> XmlElement:  # deprecated
         deprecate_with_replacement("rdfRoot", "rdf_root", "4.0.0")
         return self.rdf_root
 
-    def write_to_stream(
-        self, stream: StreamType, encryption_key: Union[None, str, bytes]
-    ) -> None:
-        self.stream.write_to_stream(stream, encryption_key)
-
-    def writeToStream(
-        self, stream: StreamType, encryption_key: Union[None, str, bytes]
-    ) -> None:  # deprecated
-        """
-        Use :meth:`write_to_stream` instead.
-
-        .. deprecated:: 1.28.0
-        """
-        deprecation_with_replacement("writeToStream", "write_to_stream", "3.0.0")
-        self.write_to_stream(stream, encryption_key)
```

Comment:
This needs to stay to help migrating users

Reply:
@MartinThoma 
this depreciation is quite old and it also applies to internal objects. I think we could try to delete it.  It will be possible to restore it later.
Also `write_to_stream()` arguments have changed and for the same reasons I think the modification can be kept

Reply:
I want to follow [our deprecation process](https://pypdf.readthedocs.io/en/latest/dev/deprecations.html):

1. Deprecation warning was added in `1.28.0` via https://github.com/py-pdf/pypdf/pull/867 in May 2022
2. It was removed in `3.0.0` <--- it should have been removed in 2.0.0
3. Originally it should be removed in 3.0.0, but now we're going to remove it in 4.0.0

It has "only" been one year since the warning was added. I want to make a `pypdf==4.0.0` release with a lot of deprecated code removals in 2024 (probably early January). Until then, I want to keep those methods to allow people to have an easier time to transition.

Is there a reason to have it in this PR?

To me it seems as if this PR would become a lot simpler without those changes.

Reply:
as @pubpub-zz said, `write_to_stream()` arguments have changed, so `writeToStream()` should change, or be deleted. and it's not used in current code, and users do not use it, it's safe to delete it.